src/js/plugins/ethereum/typedData.js

```diff
@@ -20,6 +20,17 @@ function sanitizeData(data) {
     }
 }
 
+/**
+ * Calculates the domain_separator_hash and message_hash from an EIP-712 Typed Data object.
+ *
+ * The Trezor Model 1 does not currently support constructing the hash on the device,
+ * so this function pre-computes them.
+ *
+ * @template {sigUtil.TypedMessage} T
+ * @param {T} data - The EIP-712 Typed Data object.
+ * @param {boolean} metamask_v4_compat - Set to `true` for compatibility with Metamask's signTypedData_v4 function.
+ * @returns {{domain_separator_hash: string, message_hash?: string} & T} The hashes.
+ */
```

Comment:
I felt it was overkill to setup Flow-type/Typescript just for one small file, so instead I've just written the types using Typescript JSDoc syntax: https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html

That way most IDEs should be able to see the types/documentation, but we don't have to worry about converting this file to make valid JS.